### PR TITLE
feat(adr-035): eots manager signing requests

### PR DESCRIPTION
**Pull request by Lazar955**
extends eots manager to store signing requests

[Addreses](https://github.com/babylonlabs-io/pm/blob/main/adr/adr-035-slashing-protection.md#eots-manager-protection)

### CODE REVIEW

**Comment by Lazar955** [comment]
Now `TestDoubleSigning` fails with: 
> failed to sign EOTS: rpc error: code = Unknown desc = double sign

Which is expected

File: eotsmanager/localmanager.go [hunk 1]
```diff
@@ -194,6 +199,20 @@ func (lm *LocalEOTSManager) CreateRandomnessPairList(fpPk []byte, chainID []byte
 }
 
 func (lm *LocalEOTSManager) SignEOTS(fpPk []byte, chainID []byte, msg []byte, height uint64, passphrase string) (*btcec.ModNScalar, error) {
+	record, found, err := lm.es.GetSignRecord(height)
```
Comment (gitferry) [inline eotsmanager/localmanager.go:202]: It's not safe to use data directly from db. Better to have a counter part of the db type to have validated data

**Review by gitferry — COMMENTED** [review]
Nice work! A few comments:

**Comment by gitferry** [comment]
> Which is expected. Should we now expose an UnsafeSignEOTS in which we would retain the old logic or mock EOTS for this e2e?

Good point. For this PR, we can add a case where the error is expected to make CI pass. A subsequent PR can implement `UnsafeSignEOTS` for old logic

File: eotsmanager/store/eotsstore.go [hunk 2]
```diff
@@ -93,3 +104,72 @@ func (s *EOTSStore) GetEOTSKeyName(pk []byte) (string, error) {
...
+		marshalled, err := pm.Marshal(signRecord)
+		if err != nil {
+			return err
+		}
+
+		return bucket.Put(key, marshalled)
+	})
+}
+
+func (s *EOTSStore) GetSignRecord(height uint64) (*proto.SigningRecord, bool, error) {
```
[the remark this answers is not in the document]
Comment (Lazar955) [reply eotsmanager/store/eotsstore.go:142]: Personal preference, otherwise we would have to check what type of error it is, I find this more readable

**Review by gitferry — APPROVED** [review]
Great work!

**Pull request merged**